src/sync/impl/sync_metadata.cpp

```diff
@@ -334,22 +338,31 @@ SyncUserMetadata::SyncUserMetadata(Schema schema, SharedRealm realm, RowExpr row
 , m_row(row)
 { }
 
+void SyncUserMetadata::checkValidOrThrow(const std::string operation) const
+{
+    if (!m_row.is_attached())
+        throw InvalidUserException(operation);
+}
+
 std::string SyncUserMetadata::identity() const
 {
+    checkValidOrThrow("identity");
```

Comment:
All of these checks need to happen after `verify_thread()` as that's the only thing that's safe to call from the wrong thread.